language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -381,7 +381,7 @@ data TemplateDef = TemplateDef
   -- ^ Nothing if we do not have a key.
   , tplKeyEncode :: Encode
   , tplChoices' :: [ChoiceDef]
-  , tplImplements' :: [T.Text]
+  , tplImplements' :: [(T.Text, T.Text)] -- ^ same as genTypeCon
```

Comment:
unrelated to this PR but `(T.Text, T.Text)` is a fairly useless type. We should at least throw some newtypes on these texts.